### ISSUE TITLE
Find solution for having interfaces all over the place

### ISSUE THREAD

**Issue by Fl4m3Ph03n1x** [Fl4m3Ph03n1x/market_manager#25]
Create a post in the forum explaining this issue and see how people in the community solve it.

**Comment by Fl4m3Ph03n1x** [comment]
Interesting discussion in:
https://elixirforum.com/t/how-do-you-manage-umbrella-apps-with-inner-dependencies/33962/12

**Comment by Fl4m3Ph03n1x** [comment]
Given the previous discussion had a successful conclusion, I have now moved to another discussion with more questions:

https://elixirforum.com/t/how-to-manage-files-that-have-behaviours-and-defer-to-an-implementation-at-the-same-time/34145

**Comment by Fl4m3Ph03n1x** [comment]
I have opted to follow a non-standard solution. The standard solution would be to add a Facade to every app so the app is decoupled from the other apps it communicates with. However, this would bring several disadvantages:
 - additional levels of indirection
 - would force me to use integration tests
 - would not solve the mock issues

Given I am not a fan of integration tests and I avoid them when I can, I decided to settle with alternate code version I have in the post.
